src/main/java/report/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,19 @@
+package report;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final Map<String, Integer> storage;
```

Comment:
Following the encapsulation principle, the visibility of `storage` should be private to restrict direct access from outside of the class.